robowflex_library/src/detail/cob4.cpp

```diff
@@ -0,0 +1,181 @@
+/* Author: Juan D. Hernandez */
+
+#include <cmath>
+
+#include <robowflex_library/log.h>
+#include <robowflex_library/detail/cob4.h>
+#include <robowflex_library/io.h>
+
+using namespace robowflex;
+
+const std::string  //
+    Cob4Robot::DEFAULT_URDF{"package://cob_hardware_config/robots/cob4-8/urdf/cob4-8.urdf.xacro"};
+const std::string  //
+    Cob4Robot::DEFAULT_SRDF{"package://cob_moveit_config/robots/cob4-8/moveit/config/cob4-8.srdf"};
+const std::string  //
+    Cob4Robot::DEFAULT_LIMITS{"package://cob_moveit_config/robots/cob4-8/moveit/config/joint_limits.yaml"};
+const std::string  //
+    Cob4Robot::DEFAULT_KINEMATICS{"package://cob_moveit_config/robots/cob4-8/moveit/config/kinematics.yaml"};
+const std::string  //
+    OMPL::Cob4OMPLPipelinePlanner::DEFAULT_CONFIG{"package://cob_moveit_config/robots/cob4-8/moveit/config/"
+                                                  "ompl_planning.yaml"};
+
+const std::string Cob4Robot::RESOURCE_URDF{"package://robowflex_resources/cob/robots/cob4-8.urdf.xacro"};
+const std::string Cob4Robot::RESOURCE_SRDF{"package://robowflex_resources/cob/config/cob4-8.srdf"};
+const std::string Cob4Robot::RESOURCE_LIMITS{"package://robowflex_resources/cob/config/joint_limits.yaml"};
+const std::string  //
+    Cob4Robot::RESOURCE_KINEMATICS{"package://robowflex_resources/cob/config/kinematics.yaml"};
+const std::string  //
+    OMPL::Cob4OMPLPipelinePlanner::RESOURCE_CONFIG{"package://robowflex_resources/cob/config/"
+                                                   "ompl_planning.yaml"};
+
+Cob4Robot::Cob4Robot() : Robot("cob4")
+{
+}
+
+bool Cob4Robot::initialize()
+{
+    setURDFPostProcessFunction([this](tinyxml2::XMLDocument &doc) { return addCastersURDF(doc); });
+
+    bool success = false;
+
+    // First attempt the `robowflex_resources` package, then attempt the "actual" resource files.
+    if (IO::resolvePackage(RESOURCE_URDF).empty() or IO::resolvePackage(RESOURCE_SRDF).empty())
+    {
+        RBX_INFO("Initializing Cob4 with `cob4_{description, moveit_config}`");
+        success = Robot::initialize(DEFAULT_URDF, DEFAULT_SRDF, DEFAULT_LIMITS, DEFAULT_KINEMATICS);
+    }
+    else
+    {
+        RBX_INFO("Initializing Cob4 with `robowflex_resources`");
+        success = Robot::initialize(RESOURCE_URDF, RESOURCE_SRDF, RESOURCE_LIMITS, RESOURCE_KINEMATICS);
+    }
+
+    loadKinematics("arm_left");
+    loadKinematics("arm_right");
+
+    Cob4Robot::openGrippers();
+
+    return success;
+}
+
+bool Cob4Robot::addCastersURDF(tinyxml2::XMLDocument &doc)
+{
+    for (const auto name : {"bl_caster", "br_caster", "fl_caster", "fr_caster"})
```

Comment:
This should probably be changed for the Cob4? I doubt its the exact same as the fetch.

Reply:
Done!